WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -0,0 +1,254 @@
+import SwiftUI
+
+struct WooShippingAddPackageView: View {
+    enum PackageProviderType: CaseIterable {
+        case custom, carrier, saved
+        var name: String {
+            switch self {
+            case .custom:
+                return Localization.custom
+            case .carrier:
+                return Localization.carrier
+            case .saved:
+                return Localization.saved
+            }
+        }
+    }
+
+    @Environment(\.presentationMode) var presentationMode
+
+    @State var selectedPackageType = PackageProviderType.custom
+    var addPackageButtonDisabled: Bool {
+        for (_, value) in fieldValues {
+            if value.isEmpty {
+                return true
+            }
+        }
+        return fieldValues.count != WooShippingAddPackageDimensionView.DimensionType.allCases.count
+    }
+
+    var body: some View {
+        NavigationView {
+            VStack {
+                Picker("", selection: $selectedPackageType) {
+                    ForEach(PackageProviderType.allCases, id: \.self) {
+                        Text($0.name)
+                    }
+                }
+                .pickerStyle(.segmented)
+                .padding()
+                Spacer()
+                selectedPackageTypeView
+                Spacer()
+                Button(Localization.addPackage) {
+                    // add package
+                }
+                .buttonStyle(PrimaryButtonStyle())
+                .disabled(addPackageButtonDisabled)
+                .padding()
+            }
+            .toolbar {
+                ToolbarItem(placement: .topBarLeading) {
```

Comment:
nit: What do you think of using the `.cancellationAction` placement here? It's effectively the same but has added clarity about the expected action.